appstore/developer_portal_api.py

```diff
@@ -0,0 +1,629 @@
+from .models import Category, db, App, Developer, Release, CompanionApp, Binary, AssetCollection, LockerEntry, UserLike
+from .pbw import PBW, release_from_pbw
+from .s3 import upload_pbw, upload_asset
+from .settings import config
+from .discord import announce_release, announce_new_app
+
+
+parent_app = None
+devportal_api = Blueprint('devportal_api', __name__)
+CORS(devportal_api)
+
+category_map = {
+                'Notifications': '5261a8fb3b773043d5000001',
+                'Health & Fitness': '5261a8fb3b773043d5000004',
+                'Remotes': '5261a8fb3b773043d5000008',
+                'Daily': '5261a8fb3b773043d500000c',
+                'Tools & Utilities': '5261a8fb3b773043d500000f',
+                'Games': '5261a8fb3b773043d5000012',
+                'Index': '527509e36526cda2d4000019',
+                'Faces': '528d3ef2dc7b5f580700000a',
+                'GetSomeApps': '52ccee3151a80d28e100003e',
+            }
+
+if config['ALGOLIA_ADMIN_API_KEY']:
+    algolia_client = algoliasearch.Client(config['ALGOLIA_APP_ID'], config['ALGOLIA_ADMIN_API_KEY'])
+    algolia_index = algolia_client.init_index(config['ALGOLIA_INDEX'])
+else:
+    algolia_index = None
+
+@devportal_api.route('/onboard', methods=['POST'])
+def create_developer():
+        try:
+            req = request.json
+        except Exception as e:
+            print(e)
+            return jsonify(error = "Invalid POST body. Expected JSON", e = "body.invalid"), 400
+
+        if req is None:
+            return jsonify(error = "Invalid POST body. Expected JSON", e = "body.invalid"), 400
+
+        result = authed_request('GET', f"{config['REBBLE_AUTH_URL']}/api/v1/me/pebble/appstore")
+        if result.status_code != 200:
+            abort(401)
+        me = result.json()
+
+        if not "name" in req:
+            return jsonify(error = "Missing required field: name", e = "missing.field.name"), 400
+        
+        try:
+            dev = Developer.query.filter_by(id=me['id']).one()
+            return jsonify(success = True, message = "User is already onboard")
+        except NoResultFound:
+            dev = Developer(id=me['id'], name=req['name'])
+            db.session.add(dev)
+            db.session.commit()
+
+        return jsonify(success = True, id = me['id'], message = "Onboarded user")
+
+@devportal_api.route('/submit', methods=['POST'])
+def submit_new_app():
+    try:
+
+        # Validate all fields
+        requestOK = validate_new_app_fields(request)
+
+        if requestOK[0] == True:
+
+            params = dict(request.form)
+
+            screenshots = {
+                "aplite": [],
+                "basalt": [],
+                "chalk": [],
+                "diorite": []
+            }
+
+            try:
+                pbw_file = request.files['pbw'].read()
+                pbw = PBW(pbw_file, 'aplite')
```

Comment:
`request.files` contains file-like objects, which `zipfile.ZipFile` can make use of directly.

What you are currently doing is taking a file-like object, reading all the bytes out of it, then passing it to a function that will take those bytes, use them to construct a file-like object, and read that.

Consider instead just:

```python
pbw = PBW(request.files['pbw'], 'aplite')
```

and then removing the `BytesIO` stuff from `PBW` so it actually accepts a file-like object as documented.